### DISCUSSION TITLE
Having chain problems with Google Colab

### DISCUSSION THREAD

**Discussion by oajilore** [lnccbrown/HSSM#330] [Q&A] [answered]
Is anyone using HSSM in Google Colab? 

f so, how do you make sure Jax has access to more than one device? 

I keep getting the following error. Any help is greatly appreciated!

/usr/local/lib/python3.10/dist-packages/pymc/sampling/jax.py:678: UserWarning: There are not enough devices to run parallel chains: expected 4 but got 1. Chains will be drawn sequentially. If you are running MCMC in CPU, consider using `numpyro.set_host_device_count(4)` at the beginning of your program. You can double-check how many devices are available in your system using `jax.local_device_count()`.

**Comment by digicosmos86** [comment, marked as answer]
Is your Colab using GPU? If so, then there will only be one GPU available, and JAX will use GPU, so sampling will be done sequentially. I think you can do MCMC on CPU on a CPU node and have two chains running in parallel.
